src/components/InventoryDetail/ApplicationDetails.js

```diff
@@ -44,10 +51,24 @@ const ApplicationDetails = ({
     }
   }, [disabledApps]);
 
-  const isDisconnected = verifyCulledInsightsClient(
-    getFact('per_reporter_staleness', entity)
+  const isDisconnected = useMemo(
+    () => verifyCulledReporter(perReporterStaleness, REPORTER_PUPTOO),
+    [currentApp]
   );
 
+  const isRHSMSystem = useMemo(() => {
+    const rhsmConduit =
+      !!perReporterStaleness?.[REPORTER_RHSM_CONDUIT] &&
+      !verifyCulledReporter(perReporterStaleness, REPORTER_RHSM_CONDUIT);
+    const rhsmBridge = !!perReporterStaleness?.[REPORTER_RHSM_PROFILE_BRIDGE];
+    return rhsmBridge && rhsmConduit;
+  }, [currentApp]);
+
+  const isEmptyState = (currentApp) =>
+    (currentApp === 'advisor' && isDisconnected) ||
+    (currentApp === 'vulnerabilities' && !isRHSMSystem && isDisconnected) ||
```

Comment:
I would probably put the apps names to constants so that there is less chance something will get unsynced in terms of naming